### ISSUE TITLE
getModelFromMetaCyc makes grRules with inconsistent data format

### ISSUE THREAD

**Issue by edkerk** [SysBioChalmers/RAVEN#50]
Fixes:
- [x] grRules as cell array of strings (see this comment)
- [x] consistent use of ()-brackets in grRules (see 2 comments below)

After running getModelFromMetaCyc, the resulting model.grRules uses inconsistent data formats.

grRules should be "Column Cell Array of Strings", but the current resulting cell array is a combination of strings and cells:

```matlab
>> metaCycModel = getModelFromMetaCyc();
>> metaCycModel.grRules(15:17)

ans =

  3×1 cell array

    '(MONOMER-15937)'
    {1×1 cell}
    ''
```

This should all be formatted as string (metaCycModel.grRules(15) is correct). Incorrect formatting makes other functions fail when using the metaCycModel structure.

**Comment by Hao-Chalmers** [comment]
Will look into this

**Comment by edkerk** [comment]
Changing brackets from {} to () in line 57:
`	metaCycModel.grRules{i}='';`
and line 119:
`			metaCycModel.grRules{i}=grRule;`

seems to prevent this, but it introduces problems with some other grRules (metaCycModel.grRules(15) now fails while running getMetaCycModel). So probably best to go through the code of the function and double check that, whatever the grRule defined, it always writes a string for each reaction (resulting in an array of strings).

While looking at that code, why do some reactions end up being surrounded by () and others not?

```matlab
metaCycModel.grRules(3)

ans =

  cell

    '(AT5G34850-MONOMER) or MONOMER-15160 or MONOMER-15162'
```

**Comment by Hao-Chalmers** [comment]
I don't remember exactly. the ones with () might be complexes.

**Comment by Hao-Chalmers** [comment]
MetaCyc incorporates complex-subunit relationship into associated enzymes. This information was attempted to be presented in grRule. To separate enzymes with and without subunits, I added () only to those enzymes with subunits, which means the isoenzymes without () have no subunits.

Since some complexes have only one subunits (maybe in homodimer or homotrimer). They appeared to be a bit strange.

**Comment by Hao-Chalmers** [comment]
The parentheses surrounding single-subunit enzyme complexes have been removed. Now both issues had been fixed in the branch [getModelFromMetaCyc_inconsistent_grRules_format](https://github.com/SysBioChalmers/RAVEN/compare/master...fix/getModelFromMetaCyc_inconsistent_grRules_format). please review the pull request @edkerk

**Comment by phemmy2k2** [comment]
I combined metacyc and kegg models, then I discovered I can not apply the canProduce and gapReport function on the combined model. The following is the output error.

>> CombinedDraftModel=combineMetaCycKEGGModels(model_MetaCyc, balancedModel);

>> I=canProduce(CombinedDraftModel);
Matrix index is out of range for deletion.

Error in removeReactions (line 68)
            reducedModel.grRules(indexesToDelete,:)=[];

Error in simplifyModel (line 148)
            reducedModel=removeReactions(reducedModel,rxnsToDelete);

Error in haveFlux (line 45)
smallModel=simplifyModel(model,false,false,true,true);

Error in canProduce (line 25)
produced=haveFlux(model,10^-5,rxns);
 
>> [noFluxRxns, noFluxRxnsRelaxed, subGraphs, notProducedMets, minToConnect,...
    neededForProductionMat]=gapReport(CombinedDraftModel);
Gap analysis for COMBINED - Combined model from MetaCyc and KEGG draft models

Matrix index is out of range for deletion.

Error in removeReactions (line 68)
            reducedModel.grRules(indexesToDelete,:)=[];

Error in simplifyModel (line 148)
            reducedModel=removeReactions(reducedModel,rxnsToDelete);

Error in haveFlux (line 45)
smallModel=simplifyModel(model,false,false,true,true);

Error in gapReport (line 66)
I=haveFlux(model);
